testng-core/src/main/java/org/testng/internal/ConfigurationGroupMethods.java

```diff
@@ -102,9 +107,17 @@ private Map<String, List<ITestNGMethod>> initializeAfterGroupsMap() {
     return result;
   }
 
-  public List<ITestNGMethod> getBeforeGroupMethodsForGroup(String group) {
+  public List<ITestNGMethod> getBeforeGroupMethodsForGroup(String[] groups) {
+    if (groups.length == 0) {
+      return Collections.emptyList();
+    }
+
     synchronized (beforeGroupsThatHaveAlreadyRun) {
-      return retrieve(beforeGroupsThatHaveAlreadyRun, m_beforeGroupsMethods, group);
+      return Arrays.stream(groups)
+          .map(t -> retrieve(beforeGroupsThatHaveAlreadyRun, m_beforeGroupsMethods, t))
+          .filter(Objects::nonNull)
```

Comment:
Is this `nonNull()` condition required here because I am not sure if `retrieve()` method is going to return a null value at all.

Reply:
This logic moved from initial code
`      List<ITestNGMethod> methods =
          arguments.getGroupMethods().getBeforeGroupMethodsForGroup(group);
      if (methods != null) {
        filteredMethods.addAll(methods);
      }`
Have not checked if null is possible here, just moved logic as-is

Reply:
Can you please have that checked and remove this null check which to me kind of sounds redundant now.

Reply:
we can get `null` as result of `retrieve` if there are no before/after group methods for requested group. 
So removing null check causes multiple test failures